### PR TITLE
Fix patches for emacs=28.1 build.

### PR DESCRIPTION
**Pull request by erykoff**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by erykoff** [comment]
Well, the good news is that it got past the patches.  The bad news is that something more difficult is going wrong.  I should have some time in the next few days to investigate, but this is more than a quick fix unfortunately, unless somebody else see what's going on with the bad symbol...

**Comment by asmeurer** [comment]
No idea. That's not caused by your patches is it?

I tried building it locally, but conda build just gives me the error `TypeError: Object of type date is not JSON serializable`, and I have no idea how to fix it.

**Comment by erykoff** [comment]
I tried building locally (via debug, maybe I messed something up), but that gave me an AppKit could not be run error.  And only one of the patches is applied to native osx-64 builds, and that has to do with code signing, so shouldn't be relevant here.  But I will keep digging.

**Comment by erykoff** [comment]
A-ha! man page for `openat` says `The openat() function was introduced in OS X 10.10`, and we build `osx_64` by default to a 10.9 target, and `osx_arm64` to an 11.0 target so that builds.  We can work around this, give me a moment.

**Comment by erykoff** [comment]
FYI, I'm following https://conda-forge.org/docs/maintainer/knowledge_base.html#requiring-newer-macos-sdks .  First trying the newer SDK, if that doesn't work, I'll bump the deployment target and update the run requirements.

**Comment by asmeurer** [comment]
By the way, we might want to try using the new --with-native-compilation configure flag (see https://www.masteringemacs.org/article/speed-up-emacs-libjansson-native-elisp-compilation). But maybe it would be best to do that in a later PR as it might not be straightforward.

**Comment by erykoff** [comment]
I don't see `libgccjit` or anything like that on conda forge, so that sounds like a much heavier lift.

**Comment by erykoff** [comment]
If this doesn't work, I think we need all-of-the-above (sdk, deployment, config flag).

**Comment by erykoff** [comment]
Progress, but the post-link script is now not working.  Is it possible to download artifacts from the azure builds to see what the layout is?

**Comment by erykoff** [comment]
Look at that, it builds!  I think I need to ping @conda-forge/core to confirm that we're allowed to bump the macos version number.

File: recipe/build.sh [hunk 1]
```diff
@@ -5,6 +5,12 @@ cp $BUILD_PREFIX/share/gnuconfig/config.* ./build-aux
 
 if [ "$(uname)" == "Darwin" ]; then
     OPTS=""
+    if [[ $target_platform == "osx-64" ]]; then
+        # This magic value corresponds to macos 10.10, see
+        # https://github.com/emacs-mirror/emacs/blob/575c3beb4c001687ce7a4581de005a16d6f2e081/nextstep/INSTALL#L48
+        export CFLAGS="${CFLAGS} -DMAC_OS_X_VERSION_MIN_REQUIRED=101200"
+    fi
+
```
Comment (isuruf) [inline recipe/build.sh]: Is this needed?

Comment (erykoff) [reply recipe/build.sh]: We'll find out in 15 minutes...

**Comment by asmeurer** [comment]
All green. Is this ready to go?

**Comment by erykoff** [comment]
From my perspective, it is ready.  Though I won't be able to test the osx-arm64 version until it is merged and distributed.

**Comment by asmeurer** [comment]
OK, let's try it.

**Pull request merged**